decontamination-authentication.qmd

```diff
@@ -0,0 +1,359 @@
+
+![](assets/images/chapters/decontamination-authentication/deCOM.png)
+
+*decOM* uses [kmtricks](https://academic.oup.com/bioinformaticsadvances/article/2/1/vbac029/6576015) to compute a matrix of k-mer counts on raw reads (fastq-files) from source samples, and then uses the source k-mer abundance matrix for looking up k-mer composition of sink samples. This allows *decOM* to calculate microbial contributions / fractions from the sources. For example, for estimating contributions from ancient Oral (aOral), modern Oral (mOral), Skin and Sediment / Soil environments one can use an already computed source matrix from here https://github.com/CamilaDuitama/decOM/ and provide it as a *-p_sources* parameter. Then *decOM* can be run using the following command line:
+
+```bash
+decOM -p_sources decOM_sources/ -p_sinks FASTQ_NAMES_LIST.txt \
+-p_keys decOM/FASTQ -mem 900GB -t 15
+```
+
+In the command line above, he *-p_sinks* parameter provides a list of sink samples, for example *SRR13355807*.
+The sink fastq-files are placed in *decOM/FASTQ* together with *keys* fof-files containing the mapping between fastq file names and locations of the fastq-files, for example *SRR13355807 : decOM/FASTQ/SRR13355807.fastq.gz*. The contributions from the sources to the sink samples, which are recorded in the *decOM_output.csv* output file, can then be processed and plotted as follows:
+
+```R
+df<-read.csv("decOM_output.csv", check.names=FALSE)
+
+result <- subset(df, select = c("Sink", "Sediment/Soil", "Skin", 
+"aOral", "mOral", "Unknown"))
+rownames(result) <- result$Sink
+result$Sink <- NULL
+result <- result / rowSums(result)
+result<-result[order(-result$aOral),]
+head(result)
+
+name <- rep(rownames(result), each = 5)
+value <- as.numeric(t(result))
+condition <- rep(c("Sediment/Soil","Skin","aOral","mOral","Unknown"), 
+dim(result)[1])
+data <- data.frame(name, condition, value)
+head(data)
+
+
+library("ggplot2")
+library("viridis")
+ggplot(data, aes(fill=condition, y=value, x=reorder(name,seq(1:length(name))))) + 
+  geom_bar(position = "fill", stat = "identity") + 
+  theme(axis.text.x = element_text(angle=90, size = 5, hjust = 1, vjust = 0.5)) + 
+  xlab("Sample") + ylab("Fraction")
+```
+
+![](assets/images/chapters/decontamination-authentication/decOM.png)
+
+*decOM* has certain advantages compared to *SourceTracker* as its is a taxonomic classification / database free approach. However, it also appears to be very sensitive to the particular training / source data set. In the example above it can be seen that the microbial source tracking of sink samples is very much dominated by the Oral community, which was the training / source data set.
```

Comment:
Missing a final summary output

It's also nice to include sections on:

- Questions to ask
- Useful literature

(see here as example: introduction-to-ngs-sequencing page)